src/components/Dropdown/V2/Dropdown.MenuContainer.tsx

```diff
@@ -217,9 +229,43 @@ export class MenuContainer extends React.PureComponent<Props> {
     items: Array<any>
     withIndex?: boolean
   }) => {
+    const {
+      allowMultipleSelection,
+      clearSelection,
+      focusItem,
+      selectionClearer,
+    } = this.props
+
+    if (allowMultipleSelection && selectionClearer) {
+      const clearerItem = {
+        value: selectionClearer,
+        id: createUniqueIDFactory('hsds-dropdown-v2-theallclearer')(),
+        label: '',
+      }
+
+      return [clearerItem].concat(items).map((item, index) => {
+        /* istanbul ignore next */
+        const indexProp = withIndex ? index : undefined
+        if (item.id === clearerItem.id) {
+          return (
+            <Item
+              {...this.getItemProps(item, indexProp)}
+              className="c-SelectionClearerItem"
```

Comment:
@tinkertrain Just a heads up, `getItemProps` returns a `className` key with some computed class names. To pass in `c-SelectionClearerItem` into the mix, you'd have to.. combine it with `getItemProps().className`

If this logic is too complicated here... perhaps we can add a `isClearerItem` prop to `<Item />`, which then adds it's own special className?

Reply:
I see, thanks!